core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+        murdererStats = allPlayerStats.get(murderer.getBukkit().getName());
+      }
+      if (event.getDamageInfo() instanceof ProjectileInfo) {
+        murdererStats.setLongestBowKill(
+            victim
+                .getState()
+                .getLocation()
+                .distance(((ProjectileInfo) event.getDamageInfo()).getOrigin()));
+        murdererStats.onMurder();
+        NMSHacks.sendHotbarMessage(
+            murderer.getBukkit().getPlayer(), murdererStats.getBasicStatsMessage());
+      }
+    }
+  }
+
+  @EventHandler
+  public void onMatchEnd(MatchFinishEvent event) {
+    Match match = event.getMatch();
+    HashMap<String, Integer> allKills = new HashMap<>();
+    HashMap<String, Integer> allKillstreaks = new HashMap<>();
+    HashMap<String, Integer> allDeaths = new HashMap<>();
+    HashMap<String, Integer> allBowshots = new HashMap<>();
+    for (MatchPlayer player : match.getPlayers()) {
+      allKills.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).kills);
+      allKillstreaks.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).killstreakMax);
+      allDeaths.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).deaths);
+      allBowshots.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).longestBowKill);
+    }
+    Component killMessage = getKillsMessage(sortStats(allKills));
+    Component killstreakMessage = getKillstreakMessage(sortStats(allKillstreaks));
+    Component deathMessage = getDeathsMessage(sortStats(allDeaths));
+    Map.Entry<String, Integer> bestBowshot = sortStats(allBowshots);
+    Component bowshotMessage = getBowshotMessage(bestBowshot);
+
+    for (MatchPlayer viewer : match.getPlayers()) {
+      viewer.sendMessage(
+          Components.fromLegacyText(
+              ComponentUtils.horizontalLineHeading(
+                  ChatColor.YELLOW + "Best stats this match",
+                  ChatColor.WHITE,
+                  ComponentUtils.MAX_CHAT_WIDTH)));
+
+      viewer.sendMessage(killMessage);
+      viewer.sendMessage(killstreakMessage);
+      viewer.sendMessage(deathMessage);
+      if (bestBowshot.getValue()
+          != 0) { // Prevent this from showing if bows are not used on the map played
+        viewer.sendMessage(bowshotMessage);
+      }
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map.Entry<String, Integer> sortStats(HashMap<String, Integer> map) {
+    Object[] a = map.entrySet().toArray();
+    Arrays.sort(
+        a,
+        (Comparator)
+            (o1, o2) ->
+                ((Map.Entry<String, Integer>) o2)
+                    .getValue()
+                    .compareTo(((Map.Entry<String, Integer>) o1).getValue()));
+    return (Map.Entry<String, Integer>) a[0];
```

Comment:
Something along the lines of:
```
return map.entrySet().stream()
  .max(Comparator.comparingInt(Map.Entry::getValue))
  .orElse(null);
````

Reply:
Why this in favour of what i wrote? Is there a standard i'm doing wrong? :)
Or does it just look prettier (Your does)

Reply:
It's easier to read ^^

Reply:
More concise (less code, same outcome)
Better readability
Avoids extra allocation of an object array.
Doesn't do weird casting from object to entry.
The complexity of finding a max or min is O(n) instead of O(n log n) that good sorting algs have
Isn't painful to read (yes, i'm mentioning readability twice, because it is double as relevant).